builtin/config.c

```diff
@@ -189,11 +191,60 @@ static void show_config_origin(struct strbuf *buf)
 	strbuf_addch(buf, term);
 }
 
+static const char *scope_to_string(enum config_scope scope) {
+	const char *local = "(local)";
+	const char *global = "(global)";
+	const char *system = "(system)";
+	const char *command_line = "(command line)";
+	const char *unknown = "(unknown)";
+	/* --local, --global, and --system work the same as --file so there's
+	 * no easy way for the parser to tell the difference when it is
+	 * setting the scope, so we use our information about which options
+	 * were passed
+	 */
+	if (use_local_config) {
+		return local;
+	} else if (use_global_config) {
+		return global;
+	} else if (use_system_config) {
+		return system;
+	} else {
+		switch (scope) {
+		case CONFIG_SCOPE_SYSTEM:
+			return system;
+		case CONFIG_SCOPE_GLOBAL:
+			return global;
+		case CONFIG_SCOPE_REPO:
+			return local;
+		case CONFIG_SCOPE_CMDLINE:
+			return command_line;
+		case CONFIG_SCOPE_UNKNOWN:
+		default:
+			return unknown;
+		}
+	}
+}
+
+static void show_config_scope(struct strbuf *buf)
+{
+	const char term = end_null ? '\0' : '\t';
```

Comment:
Just because the existing code is buggy does not mean you have to retain the bug when copy-editing it. So please use `end_nul` instead of `end_null`: `NUL` is a character, `NULL` is a pointer. They are totally different things.

Reply:
I changed the variable to be `end_nul` but I didn't change the option `--null` because that would probably break a lot of things